server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+	var schema models.Schema
+	filter := bson.M{"name": schemaName}
+	err := schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil {
+		return err
+	}
+	schemaVersions := schema.Versions
+	var schemaVersion models.SchemaVersion
+	for _, version := range schemaVersions {
+		filterVersion := bson.M{"_id": version}
+		err = schemasVersionCollection.FindOneAndDelete(context.TODO(), filterVersion).Decode(&schemaVersion)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = schemasCollection.DeleteOne(context.TODO(), filter)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
+	var body models.CreateNewSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.Name)
+	err := validateSchemaName(schemaName)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	exist, _, err := IsSchemaExist(schemaName)
+	if exist {
+		serv.Warnf("Schema with that name already exists")
+	}
+	if err != nil {
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("CreateNewSchema error: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
+	}
+	schemaType := strings.ToLower(body.Type)
+	err = validateSchemaType(schemaType)
+
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	schemaContent := strings.ToLower(body.SchemaContent)
+	err = validateSchemaContent(schemaContent)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	filter := bson.M{"name": schemaName}
+	var schema models.Schema
+	err = schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil && !strings.Contains(err.Error(), "no documents") {
+		serv.Errorf("CreateSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	versionNumber := len(schema.Versions) + 1
+	var active bool
+	if versionNumber == 1 {
+		active = true
+	} else {
+		active = false
+	}
+	newSchemaVersion := models.SchemaVersion{
+		ID:            primitive.NewObjectID(),
+		VersionNumber: versionNumber,
+		Active:        active,
+		CreatedByUser: user.Username,
+		CreationDate:  time.Now(),
+		SchemaContent: schemaContent,
+	}
+	newSchema := models.Schema{
+		ID:        primitive.NewObjectID(),
+		Name:      schemaName,
+		Type:      schemaType,
+		Versions:  schema.Versions,
+		IsDeleted: false,
+	}
+
+	filter = bson.M{"name": newSchema.Name, "is_deleted": false}
+	update := bson.M{
+		"$setOnInsert": bson.M{
+			"_id":  newSchema.ID,
+			"name": newSchema.Name,
+			"type": newSchema.Type,
+		},
+		"$push": bson.M{"versions": newSchemaVersion.ID},
```

Comment:
$push is wrong here you need to set the array of version to an array with single version